rebuild/src/addon.cc

```diff
@@ -0,0 +1,90 @@
+#include "addon.h"
+
+bool is_zero_bytes(
+    const uint8_t *data, const size_t start_byte, const size_t byte_length) {
+    for (size_t i = start_byte; i < byte_length; i++) {
+        if (data[i] != 0) {
+            return false;
+        }
+    }
+    return true;
+}
+
+bool is_valid_length(
+    std::string &error_out,
+    size_t byte_length,
+    size_t length1,
+    size_t length2) {
+    if (byte_length == length1 || (length2 != 0 && byte_length == length2)) {
+        return true;
+    }
+    error_out.append(" must be ");
+    if (length1 != 0) {
+        error_out.append(std::to_string(length1));
+    };
+    if (length2 != 0) {
+        if (length1 != 0) {
+            error_out.append(" or ");
+        }
+        error_out.append(std::to_string(length2));
+    };
+    error_out.append(" bytes long");
+    return false;
+}
+
+BlstTsAddon::BlstTsAddon(Napi::Env env, Napi::Object exports)
```

Comment:
Do the parameters need to be copied? Or are they just wrapped pointers, and is creating a copy trivial?

Reply:
Yes.  This is an idiosyncrasy of native node.  Those objects are reference wrappers around the values and passing them by value is necessary for the ref counter to be properly incremented/decremented. `Napi::Value`s are sort of like `std::shared_ptr` but without the atomic for synchronization because its single threaded...